www/nginx/src/opnsense/service/templates/OPNsense/Nginx/nginx.conf

```diff
@@ -33,4 +33,8 @@ stream {
 {#   include http blocks partial #}
 {%   include "OPNsense/Nginx/mail.conf" ignore missing with context %}
 # }
+
+# Map used in location.conf for proxy_ssl_name
+map $ssl_server_name $upstream_sni_name { default $ssl_server_name; '' $host; }
```

Comment:
http.conf template already have map directive with a proper nginx.conf style (https://github.com/opnsense/plugins/blob/548566b52db2e7a5799bb67530f88086b63546b6/www/nginx/src/opnsense/service/templates/OPNsense/Nginx/http.conf#L50). i would place it there, after the $connection_upgrade mapping